src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcherLoadBalancer.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    internal class FunctionDispatcherLoadBalancer : IFunctionDispatcherLoadBalancer
+    {
+        private static int _counter = 0;
+
+        internal int Counter => _counter;
+
+        public ILanguageWorkerChannel GetLanguageWorkerChannel(IEnumerable<ILanguageWorkerChannel> languageWorkerChannels, int maxProcessCount)
+        {
+            if (languageWorkerChannels == null)
+            {
+                throw new ArgumentNullException(nameof(languageWorkerChannels));
+            }
+
+            var currentNumberOfWorkers = languageWorkerChannels.Count();
```

Comment:
Yes. We need to pass the channels collection here as it can change at any point of time. For example, while restarting there will be one process to start with. Rest of the processes are debouce-started